autogen.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```

Comment:
I don't see the point of having `autogen.sh` files in every project repository. Instead of:

```sh
./autogen.sh
```

I can just do:

```sh
autoreconf [-fiv]
```

and have less cruft in the repository. Also, If there is no `autogen.sh` JHBuild will automatically use `autoreconf && ./configure ...` instead, so no harm done :wink: